pkg/extended_registry/ghcr.go

```diff
@@ -0,0 +1,302 @@
+package extendedregistry
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/aserto-dev/go-grpc/aserto/api/v1"
+	"github.com/google/go-containerregistry/pkg/authn"
+	"github.com/google/go-containerregistry/pkg/name"
+	"github.com/google/go-containerregistry/pkg/v1/remote"
+	"github.com/pkg/errors"
+	"github.com/rs/zerolog"
+	"google.golang.org/protobuf/types/known/timestamppb"
+)
+
+type GHCRClient struct {
+	base *xClient
+}
+
+var images []struct {
+	Name       string `json:"name"`
+	Visibility string `json:"visibility"`
+	Owner      struct {
+		Login string `json:"login"`
+	}
+}
+
+type ContainerInfo struct {
+	Tags []string `json:"tags"`
+}
+type Metadata struct {
+	PackageType string `json:"package_type"`
+	Container   ContainerInfo
+}
+type TagResponse struct {
+	ID        int      `json:"ID"`
+	SHA       string   `json:"name"`
+	CreatedAt string   `json:"created_at"`
+	Meta      Metadata `json:"metadata"`
+}
+
+func NewGHCRClient(logger *zerolog.Logger, cfg *Config, client *http.Client) ExtendedClient {
+	baseClient := newExtendedClient(logger, cfg, client)
+
+	return &GHCRClient{
+		base: baseClient,
+	}
+}
+
+func (g *GHCRClient) ListOrgs() ([]string, error) {
+	orgsresponse, err := g.base.get("https://api.github.com/user/orgs")
```

Comment:
maybe we could use the gihub SDK: https://github.com/google/go-github instead of making http requests